dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java

```diff
@@ -52,68 +62,117 @@ public static void copyAppToDropins() throws Exception {
 
     @Before
     public void setUp() throws Exception {
-        TestMethod testMethod = getTestMethod(TestMethod.class, testName);
-        configureBeforeCheckpoint(testMethod);
+        testMethod = getTestMethod(TestMethod.class, testName);
+        configureBeforeCheckpoint();
         server.setCheckpoint(CheckpointPhase.APPLICATIONS, true,
                              server -> {
                                  assertNotNull("'SRVE0169I: Loading Web Module: " + APP_NAME + "' message not found in log before rerstore",
                                                server.waitForStringInLogUsingMark("SRVE0169I: .*" + APP_NAME, 0));
                                  assertNotNull("'CWWKZ0001I: Application " + APP_NAME + " started' message not found in log.",
                                                server.waitForStringInLogUsingMark("CWWKZ0001I: .*" + APP_NAME, 0));
-                                 configureBeforeRestore(testMethod);
+                                 configureBeforeRestore();
                              });
         server.startServer(getTestMethod(TestMethod.class, testName) + ".log");
     }
```

Comment:
It's fairly difficult to work out what's going on in this test, you need to check several different places for each test method to find out how it's set up and what it's trying to do. It would be good to find some way to completely refactor this test so that you can see for an individual test method what environment variables are set before and after checkpoint, what log messages are expected and what the application code is doing in one place rather than split out all over the place.

Reply:
Given that you're starting and stopping the server for each test, you could just have a method for each test which does the configuration, starts the server, checkpoints, calls the test servlet, stops the server and checks the logs.